modules/code-editor/src/backend/editor.ts

```diff
@@ -75,6 +75,46 @@ export default class Editor {
     }
   }
 
+  async deleteFile(file: EditableFile): Promise<void> {
+    this._validateMetadata(file)
+    const { location, botId, hookType } = file
+    const ghost = botId ? this.bp.ghost.forBot(this._botId) : this.bp.ghost.forGlobal()
+
+    if (file.type === 'action') {
+      return ghost.deleteFile('/actions', location)
+    }
+    if (file.type === 'hook') {
+      return ghost.deleteFile(`/hooks/${hookType}`, location.replace(hookType, ''))
+    }
+  }
+
+  async renameFile(file: EditableFile, newName: string): Promise<EditableFile> {
+    this._validateMetadata(file)
```

Comment:
There is no validation of the new name. If it has invalid characters, the file will be deleted but not re-created. (using renameFile instead would avoid that kind of thing)